client/landing/stepper/declarative-flow/internals/components/step-route/hooks/use-step-route-tracking/index.tsx

```diff
@@ -50,7 +68,7 @@ export const useStepRouteTracking = ( {
 	useEffect( () => {
 		// We record the event only when the step is not empty. Additionally, we should not fire this event whenever the intent is changed
 		if ( ! hasRequestedSelectedSite || skipTracking ) {
-			return;
+			return () => handleRecordStepComplete( { flowName, stepSlug, intent } );
 		}
```

Comment:
@daledupreez can you please confirm why we would skip tracking a step? Irrespective of content, shouldn't we still track it as normal if it renders?